tools/importer-rest-api-specs/components/parser/helpers_test.go

```diff
@@ -37,6 +37,20 @@ func ParseSwaggerFileForTesting(t *testing.T, file string, serviceName *string)
 		t.Fatalf("parsing file %q: %+v", file, err)
 	}
 
+	// removeUnusedItems used to be called as we iterated through the swagger files
+	// it's now called once after all the processing for a service has been done so must be called here
+	// to replicate the entire parsing process for swagger files
+	for resource, details := range out.Resources {
+		resourceModels, resourceConstants, resourceIdsToUri := removeUnusedItems(details.Operations, details.ResourceIds, details.Models, details.Constants)
+		out.Resources[resource] = importerModels.AzureApiResource{
+			Constants:   resourceConstants,
+			Models:      resourceModels,
+			Operations:  details.Operations,
+			ResourceIds: resourceIdsToUri,
+			Terraform:   details.Terraform,
+		}
+	}
```

Comment:
might seem pedantic, but could we make this (and the usage below) call the same method, which takes `out.Resources` as an argument? that'd mean we don't have these unintentionally fall out of sync